src/proto/h2/client.rs

```diff
@@ -128,7 +128,7 @@ where
                     let (head, body) = req.into_parts();
                     let mut req = ::http::Request::from_parts(head, ());
                     super::strip_connection_headers(req.headers_mut(), true);
-                    if let Some(len) = body.content_length() {
+                    if let Some(len) = body.size_hint().upper() {
```

Comment:
If the lower hint isn't the same, then we can't know the actual content-length, right?

Reply:
done